packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2227,39 +2151,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.dataStores.createDetachedDataStoreCore(pkg, false);
     }
 
-    /**
-     * Creates a possibly root datastore directly with a possibly user generated id and attaches it to storage.
-     * It is vulnerable to name collisions if both aforementioned conditions are true, and should not be used.
-     *
-     * This method will be removed. See #6465.
-     */
-    private async _createDataStoreWithPropsLegacy(
+    public async _createDataStoreWithProps(
         pkg: string | string[],
         props?: any,
         id = uuid(),
         isRoot = false,
     ): Promise<IDataStore> {
-        const fluidDataStore = await this.dataStores._createFluidDataStoreContext(
-            Array.isArray(pkg) ? pkg : [pkg], id, isRoot, props).realize();
         if (isRoot) {
-            fluidDataStore.makeVisibleAndAttachGraph();
-            this.logger.sendTelemetryEvent({
-                eventName: "Root datastore with props",
-                hasProps: props !== undefined,
-            });
+            throw new UsageError("Creating root datastores is not supported. Use datastore aliasing instead");
```

Comment:
If this method is never supposed to receive `isRoot = true`, can we just remove the parameter altogether? If it's too late to do it in 2.x-internal-1.0 because it wasn't announced, maybe in 2.x-internal-2.0.

Reply:
I would do it by removing the function altogether as this was deprecated years ago. The exception is intentional. Hiding this by removing the parameter would lead to some bad outcomes where the datastore is supposed to be root but it's not and risks getting gc'ed, which would be a pain to debug.